### ISSUE TITLE
Where are the results of semantic segmentation stored?

### ISSUE THREAD

**Issue by gyy520cyaowu** [drprojects/superpoint_transformer#96]
Where are the results of semantic segmentation stored?I want to use nag.load() to load my segmentation results (I don’t know if this is correct), but I can’t find the required .pt file, only the demo in notebooks.I don’t know where the input required in the parameter list of those tools is.

**Comment by drprojects** [comment]
The outputs of the semantic segmentation are not stored anywhere. `NAG.load()` will load the data preapred at preprocessing time. Then the provided pipeline applies some transforms and a forward pass of the model to produce the predictions. 

PS: if you ❤️ or use this project, don't forget to give it a ⭐, it means a lot to us !

**Comment by gyy520cyaowu** [comment]
> The outputs of the semantic segmentation are not stored anywhere. `NAG.load()` will load the data preapred at preprocessing time. Then the provided pipeline applies some transforms and a forward pass of the model to produce the predictions.
> 
> PS: if you ❤️ or use this project, don't forget to give it a ⭐, it means a lot to us !

那我该怎么查看我的分割效果

**Comment by narges-tk** [comment]
Likewise, I am wondering how I can visualize/ get the segmented point cloud. Actually, I used pre-trained model and now evaluation went well. But where is exactly the segmented point cloud or how can I implement this part it if it is not already done?

**Comment by drprojects** [comment]
Computing and visualizing point clouds -> [`notebooks/demo.ipyng`](https://github.com/drprojects/superpoint_transformer/blob/master/notebooks/demo.ipynb).

Where/what is the semantic segmentation output ? -> [`notebooks/demo.ipyng`](https://github.com/drprojects/superpoint_transformer/blob/master/notebooks/demo.ipynb) and [`SemanticSegmentationOutput`](https://github.com/drprojects/superpoint_transformer/blob/master/src/utils/output_semantic.py).

**Comment by gyy520cyaowu** [comment]
> Likewise, I am wondering how I can visualize/ get the segmented point cloud. Actually, I used pre-trained model and now evaluation went well. But where is exactly the segmented point cloud or how can I implement this part it if it is not already done?

Sir, do you know how to visualize?

**Comment by drprojects** [comment]
@gyy520cyaowu the answer is literally right above your post... :roll_eyes:

**Comment by gyy520cyaowu** [comment]
> @gyy520cyaowu the answer is literally right above your post... 🙄

In what file format is the segmentation result stored? In other similar projects, I always find the txt document in exp/

**Comment by narges-tk** [comment]
> > @gyy520cyaowu the answer is literally right above your post... 🙄
> 
> In the demo file you mentioned, some methods of viewing nag are introduced, including the loading method: nag = torch.load('demo_nag.pt'). Only with this nag can we perform subsequent visualization operations. However, we need to find the .pt file we split ourselves. Even if we run 100 different areas using your method, we can only see the result of demo_nag.pt, and we cannot find our own .pt file. Shouldn't we first find the .pt results we segmented before we can use your methods?

I agree with you. I need txt (or other format) files of segmentation outputs. Besides, during demo visualization, I found that it is not able to visualize the entire of my data which is just with similar number of points like used benchmark data. Any more clarification would be appreciated.

**Comment by gardiens** [comment]
> I agree with you. I need txt (or other format) files of segmentation outputs. Besides, during demo visualization, I found that it is not able to visualize the entire of my data which is just with similar number of points like used benchmark data. Any more clarification would be appreciated.

Hello, as said my drproject said :
"The outputs of the semantic segmentation are not stored anywhere. NAG.load() will load the data prepared at preprocessing time. Then the provided pipeline applies some transforms and a forward pass of the model to produce the predictions." 
so the outputs is not stored  and no visualization besides the nag.show is currently supported as far as I know. 
You can try to implement it yourself and open a pull request :).
 You could for instance use [open3d ](https://www.open3d.org/) or another cloud points visualizers, 
 

> In the demo file you mentioned, some methods of viewing nag are introduced, including the loading method: nag = torch.load('demo_nag.pt'). Only with this nag can we perform subsequent visualization operations. However, we need to find the .pt file we split ourselves. Even if we run 100 different areas using your method, we can only see the result of demo_nag.pt, and we cannot find our own .pt file. Shouldn't we first find the .pt results we segmented before we can use your methods?
You should look at [this part of the documentation](https://github.com/drprojects/superpoint_transformer/blob/master/docs/data_structures.md) where it explains how the data flows works

**Comment by narges-tk** [comment]
> > I agree with you. I need txt (or other format) files of segmentation outputs. Besides, during demo visualization, I found that it is not able to visualize the entire of my data which is just with similar number of points like used benchmark data. Any more clarification would be appreciated.
> 
> Hello, as said Drproject : "The outputs of the semantic segmentation are not stored anywhere. NAG.load() will load the data prepared at preprocessing time. Then the provided pipeline applies some transforms and a forward pass of the model to produce the predictions." so the outputs is not stored and no visualization besides the nag.show is currently supported as far as I know. You can try to implement it yourself and open a pull request :). You could for instance use [open3d ](https://www.open3d.org/) or another cloud points visualizers,
> 
> > In the demo file you mentioned, some methods of viewing nag are introduced, including the loading method: nag = torch.load('demo_nag.pt'). Only with this nag can we perform subsequent visualization operations. However, we need to find the .pt file we split ourselves. Even if we run 100 different areas using your method, we can only see the result of demo_nag.pt, and we cannot find our own .pt file. Shouldn't we first find the .pt results we segmented before we can use your methods?
> 
> You should look at [this part of the documentation](https://github.com/drprojects/superpoint_transformer/blob/master/docs/data_structures.md) where it explains how the data flows works

Thanks for your help!
